my-dsl.md

```diff
@@ -1,38 +1,79 @@
 # Language
-_What is the name of the language? Link the name to its webpage 
+_What is the name of the language? Link the name to its webpage
 (if appropriate)._
 
+[should.js](https://shouldjs.github.io/)
 
 # Domain
 _Describe the language's domain in five words._
 
+Descriptive unit tests for Javascript.
 
 # Computational model
-_We don't yet have a great definition of the term "computational model". 
-For now, try to come up with the clearest, most concise explanation of 
+_We don't yet have a great definition of the term "computational model".
+For now, try to come up with the clearest, most concise explanation of
 what happens when a program in your DSL runs._
 
+The language is an internal, fragmentary language, so its computational model
+is inherited from its host language Javascript. This means that when tests which
+use should.js run, the should.js code is interpreted as valid Javascript code,
+often intermixed with general Javascript, and is fed to the JS interpreter to be
+executed.
 
 # DSL-ness
-_Fowler writes about a spectrum of languages, from general-purpose languages to 
-"purely" domain-specific. Where does the DSL you chose fall on this spectrum, 
-and why?_ 
+_Fowler writes about a spectrum of languages, from general-purpose languages to
+"purely" domain-specific. Where does the DSL you chose fall on this spectrum,
+and why?_
 
+should.js is purely domain-specific. Its lack of Turing completeness is the
+biggest indicator: the langauge does not have variables, conditionals, or
+looping constructs (although all of these may be found interspersed with
+should.js code in native Javascript, due to the fragmentary nature of the DSL).
+
+The only thing should.js is useful for is making assertions about code in a
+self-documenting way.
 
 # Internal or external?
-_Is the language implemented as an internal or external DSL? 
+_Is the language implemented as an internal or external DSL?
 Justify your answer._
 
+It is internal: should.js code is valid Javascript.
 
 # Host language
 _What language(s) was (were) used to implement the DSL?_
 
+Javascript.
 
 # Benefits
-_Identify one potential benefit of the DSL: how is a programmer's life or a 
+_Identify one potential benefit of the DSL: how is a programmer's life or a
 company's bottom line made easier by the existence of this language?_
 
+should.js allows the programmer to write unit tests that not only prove the
+correctness of the program, but also document what the correct behavior should
+be. This is done in a way that is readable even if the reader does not know
+anything about the host language. For example,
+
+```javascript
```

Comment:
I appreciate this example! I've never written in Javascript before, but I understood this. :)